pkg/crc/services/dns/dns.go

```diff
@@ -109,7 +109,27 @@ func CheckCRCLocalDNSReachable(ctx context.Context, serviceConfig services.Servi
 }
 
 func CheckCRCPublicDNSReachable(serviceConfig services.ServicePostStartConfig) (string, error) {
-	stdout, _, err := serviceConfig.SSHRunner.Run(fmt.Sprintf("host -R 3 %s", publicDNSQueryURI))
+	// This does not query DNS directly to account for corporate environment where external DNS resolution
+	// may only be done on the host running the http(s) proxies used for internet connectivity
+	proxyConfig, err := network.NewProxyConfig()
+	if err != nil {
+		// try without using proxy
+		proxyConfig = &network.ProxyConfig{}
+	}
+	curlArgs := []string{"--head", publicDNSQueryURI}
+	if proxyConfig.IsEnabled() {
+		if proxyConfig.HTTPProxy != "" {
+			curlArgs = append(curlArgs, "--proxy", proxyConfig.HTTPProxy)
+		}
+		if proxyConfig.HTTPSProxy != "" {
+			curlArgs = append(curlArgs, "--proxy", proxyConfig.HTTPSProxy)
+		}
+		curlArgs = append(curlArgs, "--noproxy", proxyConfig.GetNoProxyString())
+		if proxyConfig.ProxyCAFile != "" {
+			curlArgs = append(curlArgs, "--proxy-cacert", proxyConfig.HTTPProxy)
+		}
+	}
+	stdout, _, err := serviceConfig.SSHRunner.Run("curl", curlArgs...)
```

Comment:
Can we not ignore `stderr` and club it with error because sometime stderr provide some context behind the error?

```
return stdout, fmt.Errorf("stderr:%s, err: %v", stderr, err)
```

Reply:
not blocker but will take care of in following PR

Reply:
I still haven't merged this branch https://github.com/cfergeau/crc/tree/exec-error which would solve this once and for all, but it itches me every time I see the addition of `stderr := ...` :)

Instead of copying the same code everywhere (`stdout, stderr, err := Run(...)` followed by `logging.Warnf("%s: %v", stderr, err)`), we can save the duplication by using a specialized error type: https://github.com/cfergeau/crc/commit/c6e3953afa2e532f96237030e09651ead03d0770

I keep rebasing that branch, cleaning it up a little bit, but still missing testing and the last bit of polish before filing a PR :)